### PR TITLE
Feat/reverse

### PR DESCRIPTION
**Pull request by nattsnb**

### CODE REVIEW

File: LinkedList.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  reverse = () => {
+    if (this.#tail && this.#head !== this.#tail) {
+      const arrayFromList = this.toArray();
+      arrayFromList.reverse();
+      const reversedList = new LinkedList();
+      for (const element of arrayFromList) {
+        reversedList.addToTail(element.value);
+      }
```
Comment (Solnick) [inline LinkedList.js:48]: Reversing a linked list should have O(n) complexity. 
Please do not use the `toArray` method for other purposes than debugging.

File: LinkedList.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    let currentMember = this.#head;
+    const array = [];
+    while (currentMember !== null) {
+      array.push(currentMember);
+      currentMember = currentMember.nextMember;
+    }
+    return array;
+  }
+
+  remove() {
```
Comment (Solnick) [inline LinkedList.js:33]: this function should take argument/arguments so there is a possibility to remove any element

File: LinkedList.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      const arrayFromList = this.toArray();
+      arrayFromList.reverse();
+      const reversedList = new LinkedList();
+      for (const element of arrayFromList) {
+        reversedList.addToTail(element.value);
+      }
+      return reversedList;
+    }
+  };
+  switchPositions(leftElement, rightElement) {
```
Comment (Solnick) [inline LinkedList.js:52]: have you tested it?

File: LinkedList.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    rightElement.nextMember.previousMember = leftElement;
+    leftElement.previousMember.nextMember = rightElement;
+    rightElement.previousMember = leftElement.previousMember;
+    leftElement.nextMember = rightElement.nextMember;
+    leftElement.previousMember = rightElement;
+    rightElement.nextMember = leftElement;
+  }
+  addBetween(previousElement, nextElement) {
```
Comment (Solnick) [inline LinkedList.js:60]: please add a value argument

**Review by Solnick — CHANGES_REQUESTED** [review]

File: LinkedList.js (hunk 4 above)
Comment (nattsnb) [reply LinkedList.js:60]: done

File: LinkedList.js (hunk 2 above)
Comment (nattsnb) [reply LinkedList.js:33]: done

File: LinkedList.js (hunk 3 above)
Comment (nattsnb) [reply LinkedList.js:52]: done

File: LinkedList.js (hunk 1 above)
Comment (nattsnb) [reply LinkedList.js:48]: done

**Pull request merged**